### ISSUE TITLE
inherit zoom only works for 1st level bookmarks

### ISSUE THREAD

**Issue by yippiekahee** [beatngu13/pdf-zoom-wizard#82]
**Describe the bug**
The app only sets first level bookmarks to Inherit zoom. With 2nd & 3rd level my document still jumps to Fit width or other zoom level.

**To Reproduce**
pdf is not to be made public, regretfully

**Expected behavior**
Inherit zoom on all levels

**Log**
This is a weird location, cannot find it
Add the log file from `${HOME}/.pdfzoomwizard/log.txt`.

**Desktop (please complete the following information):**
 - OS: Win 10.0.17763 Build 17763
 - Java version: 1.8.0_221
 - PDF Zoom Wizard version: 0.4.1

**Additional context**

**Comment by beatngu13** [comment]
Hey @yippiekahee,

Thank you for that detailed bug report!

Without the original or a similar PDF, it is hard to reproduce the issue. Maybe you can help me with the log: As I [described in the wiki](https://github.com/beatngu13/pdf-zoom-wizard/wiki#info), on Windows the path should be something like:

```
C:\Users\alice\.pdfzoomwizard\log.txt
```

Where "alice" is replaced by your username.

Can you find it now? If yes, please attach it to this issue _after_ you have executed the Wizard with your PDF.

**Comment by yippiekahee** [comment]
I edited chapters to bla, sorry. But clearly all higher levels are not addressed

21:31:01.115 [JavaFX Application Thread] INFO  Launcher - Launching PDF Zoom Wizard v0.4.1.
21:31:15.751 [Thread-4] INFO  Wizard - Start working on 'bla'. PDF document(s) will be saved with serialization mode 'Incremental'.
21:31:15.752 [Thread-4] INFO  Wizard - Processing 'bla.pdf'.
21:31:15.825 [Thread-4] INFO  Wizard - Modified bookmark 'bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.825 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 1: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.826 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 2: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.826 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 3: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.827 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 4: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.827 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 5: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.828 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 6: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.828 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 7: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.829 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 8: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.829 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 9: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.829 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 10: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.830 [Thread-4] INFO  Wizard - Modified bookmark 'Chapter 11: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.830 [Thread-4] INFO  Wizard - Modified bookmark 'APPENDIX A: bla' to use mode 'XYZ' and zoom 'null'.
21:31:15.830 [Thread-4] INFO  Wizard - Modified bookmark 'bla' to use mode 'XYZ' and zoom 'null'.
21:31:16.385 [Thread-4] INFO  Wizard - Modified 14 bookmark(s) in 'bla.pdf'.
21:31:16.386 [Thread-4] INFO  Wizard - Modified 14 bookmark(s) in 1 file(s).

**Comment by yippiekahee** [comment]
I should prob. say lower levels..

**Comment by beatngu13** [comment]
Thx, will try to have a look into it over the next days.

**Comment by beatngu13** [comment]
I've just tested it with the following dummy PDF:

http://www.pdf995.com/samples/pdf.pdf

As you can see, it has a nested bookmark structure, which yields this log:

```
19:54:49.885 [Thread-7] INFO  Wizard - Start working on '/Users/kruse/Downloads/pdf.pdf'. PDF document(s) will be saved with serialization mode 'Incremental'.
19:54:49.885 [Thread-7] INFO  Wizard - Processing 'pdf.pdf'.
19:54:49.892 [Thread-7] INFO  Wizard - Modified bookmark 'Introduction to the Pdf995 Suite' to use mode 'XYZ' and zoom 'null'.
19:54:49.893 [Thread-7] INFO  Wizard - Modified bookmark 'Hierarchical bookmark to Stamped Document' to use mode 'XYZ' and zoom 'null'.
19:54:49.893 [Thread-7] INFO  Wizard - Modified bookmark 'Sample Word document with links' to use mode 'XYZ' and zoom 'null'.
19:54:49.894 [Thread-7] INFO  Wizard - Modified bookmark 'Excel Chart and Spreadsheet' to use mode 'XYZ' and zoom 'null'.
19:54:49.897 [Thread-7] INFO  Wizard - Modified 4 bookmark(s) in 'pdf.pdf'.
19:54:49.897 [Thread-7] INFO  Wizard - Modified 4 bookmark(s) in 1 file(s).
```

So, it looks like nesting itself is not the issue.

Can you provide some sample PDF that allows me to reproduce the issue?

**Comment by beatngu13** [comment]
Will close this due to inactivity. Feel free to re-open if you again encounter this issue and can provide a sample PDF to reproduce the error.

**Comment by Zackhardtoname** [comment]
Does not work on [this one](http://dl.konkur.in/post/Book/Paye/Elementary-Differential-Equations-and-Boundary-Value-Problems-Boyce-11th-Edition-%5Bkonkur.in%5D.pdf)

**Comment by beatngu13** [comment]
@Zackhardtoname thx! Will have a look and re-open the issue if necessary.

**Comment by beatngu13** [comment]
@Zackhardtoname just tested it. Here's an excerpt of the log:

```text
22:40:59.841 [Thread-7] INFO  Wizard - Processing 'Elementary-Differential-Equations-and-Boundary-Value-Problems-Boyce-11th-Edition-[konkur.in].pdf'.
22:40:59.937 [Thread-7] INFO  Wizard - Modified bookmark 'Cover'.
22:40:59.937 [Thread-7] INFO  Wizard - Modified bookmark 'Title Page'.
22:40:59.938 [Thread-7] INFO  Wizard - Modified bookmark 'Copyright'.
22:40:59.938 [Thread-7] INFO  Wizard - Modified bookmark 'Dedication'.
22:40:59.938 [Thread-7] INFO  Wizard - Modified bookmark 'The Authors'.
22:40:59.939 [Thread-7] INFO  Wizard - Modified bookmark 'Preface'.
22:40:59.939 [Thread-7] INFO  Wizard - Modified bookmark 'Brief Contents'.
22:40:59.939 [Thread-7] INFO  Wizard - Modified bookmark 'Contents'.
22:40:59.940 [Thread-7] INFO  Wizard - Modified bookmark 'Problems'.
22:40:59.941 [Thread-7] INFO  Wizard - Modified bookmark '1.1. Some Basic Mathematical Models; Direction Fields'.
22:40:59.941 [Thread-7] INFO  Wizard - Modified bookmark 'Problems'.
```

As you can see, bookmarks of level 2, 3, … are modified. Also when I check the resulting PDF, the selected zoom (I tested "inherit zoom") is applied.

What PDF Zoom Wizard version are using? Have tried the newest one?

**Comment by Zackhardtoname** [comment]
I used v0.6.0. Could you kindly test on [pdfstudioviewer (free)](https://www.qoppa.com/pdfstudioviewer/)?

**Comment by beatngu13** [comment]
@Zackhardtoname yesterday I released v0.7.0, but v0.6.0 is fine as there have been no bugfixes in between.

Regarding PDF Studio Viewer: I'm sorry, I only test against Adobe Acrobat (Reader), no 3rd party PDF viewers/tools. Some of them have special internal settings that overwrite/ignore a PDF's bookmark zoom configuration.

Have you tried Acrobat Reader? To my knowledge it is free and available on all major platforms.

**Comment by Zackhardtoname** [comment]
Unfortunately I don't think Acrobat Reader is availabe on Linux unless I use wine...
